tests/test.py

```diff
@@ -1174,6 +1174,60 @@ def test_rule_expression_filetools(self):
         result = rule.evaluate(sample)
         self.assertEqual(result.result, Result.ignored)
 
+    def test_rule_expression_knowntools(self):
+        """ Test generic rule on knowntoolsreport. """
+        config = r'''[expressions]
+            expression.0  : sample.filename == "first.a" and knownreport.known -> bad
+            expression.1  : knownreport.known -> knownreport.result
```

Comment:
Whoa, that's a good one just innocently popping up there. Great stuff. At the very least it deserves a note in the commit message, perhaps some documentation. But first I'd really like to be sure, we really want to have this kind of thing. It certainly allows to nicely intermingle the the behaviours of multiple complex expressions and the traditional KnowRule. Any fallout already visible down the road?